client/product-details/index.js

```diff
@@ -7,27 +7,39 @@
 import { initializeBnplSiteMessaging } from './bnpl-site-messaging';
 
 jQuery( function ( $ ) {
+	// Check for required global variables
+	if ( ! window.wcpayStripeSiteMessaging ) {
+		return;
+	}
 	const bnplPaymentMessageElement = initializeBnplSiteMessaging();
 	const { productVariations } = window.wcpayStripeSiteMessaging;
 	let { productId } = window.wcpayStripeSiteMessaging;
 
 	const resetBnplPaymentMessage = () => {
-		const quantity = $( '.quantity input[type=number]' ).val();
+		const quantityInput = $( '.quantity input[type=number]' );
```

Comment:
Also here: if we couldn't parse new quantity or variation price and currency are not accessible, we shouldn't update the element.

Reply:
Done here https://github.com/Automattic/woocommerce-payments/commit/23da3b2d9a1c97f84b4de2c74bcd82774e53d9ef thanks